core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -339,6 +340,24 @@ object SparkSubmit {
       }
     }
 
+    // In yarn mode for an R app, add the SparkR package archive to archives
+    // that can be distributed with the job
+    if (args.isR && clusterManager == YARN) {
+      val sparkHome = sys.env.get("SPARK_HOME")
+      if (sparkHome.isEmpty) {
+        printErrorAndExit("SPARK_HOME does not exist for R application in yarn mode.")
+      }
+      val rPackagePath = Seq(sparkHome.get, "R", "lib").mkString(File.separator)
```

Comment:
Is there a way to make this also go through `RUtils.sparkRPackagePath`? It seems weird to have two separate code paths here.